openshift-ci/README.md

```diff
@@ -0,0 +1,193 @@
+`OPENSHIFT_BUILD_NAMESPACE` environment variable.
+
+## root
+
+As part of the CI pipeline, the first step is to create `root` image.
+In fact, `root` is a tag created for the pipeline image.  This image
+contains all the tools including but not limited to Go compiler, git,
+kubectl, oc, and Operator SDK.
+
+The `root` image tag is created using this Dockerfile:
+`openshift-ci/Dockerfile.tools`
+
+## src
+
+This step clone the pull request branch with latest changes.  The
+cloning is taking place inside a container created from the `root`
+image.  As a result of this step an image named `src` is going to be
+created.
+
+In the CI configuration, there is a declaration like this:
+
+```
+canonical_go_repository: github.com/redhat-developer/devconsole-operator
+```
+
+The above line ensure the cloned source code goes into the specified
+path: `$GOPATH/src/<canonical_go_repository>`.
+
+## bin
+
+This step runs the `build` Makefile target.  This step is taking place
+inside a container created from the `src` image created in the
+previous step.
+
+The `make build` produce operator binary image available under
+`./out` directory.  Later, this binary is copied to
+`devconsole-operator` (see below).
+
+As a result of this step an container image named `bin` is going to be
+created.
+
+
+## images
+
+There are three container images that are built as part of this job.
+Before this step, a couple of base images are tagged from existing
+published images.  The first one is a CentOS 7 image referred as `os`
+in the CI configuration.  The other one is the operator registry image
+which contains all the binaries to run a gRPC based registry.  The
+operator image registry is referred as `operator-registry` in the CI
+configuration.
+
+### devconsole-operator
+
+Thee CentOS 7 image (`os`) is used as the base image for creating
+`devconsole-operator` image.  The operator binary available inside
+`bin` container image is copied over here.  The Dockerfile used is
+available here: `openshift-ci/Dockerfile.deploy`
+
+The image produced can be pulled from here:
+`registry.svc.ci.openshift.org/${OPENSHIFT_BUILD_NAMESPACE}/stable:devconsole-operator`
+
+### operator-registry-base
+
+The `operator-registry` is used as the base image for creating
+`operator-registry-base` image.  This is an intermediate image used to
+propagate value of `OPENSHIFT_BUILD_NAMESPACE` environment variable to
+`devconsole-operator-registry` image build.  This intermediate image
+is going to be used as the base image for
+`devconsole-operator-registry`.  The original `operator-registry`
+image has a a different value for `OPENSHIFT_BUILD_NAMESPACE`
+environment variable, which must be set when that image was built.
+
+The Dockerfile used is available here:
+`openshift-ci/Dockerfile.registry.intermediate` As you can see, the
+Dockerfile file has only one line with a `FROM` statement, which is
+going to replaced with `operator-registry` image during image build.
```

Comment:
Should be "going to be replaced"

Reply:
Don. ceba1627b1c5d4a8ced946a3b352b1416f4e6c3a